### PR TITLE
Implement backend to save course key for already created course run.

### PR DESCRIPTION
**Pull request by waheedahmed**
[ECOM-6178](https://openedx.atlassian.net/browse/ECOM-6178)

Implement the backend to save course run key in pending studio instances and send an email to course team with subject "Studio instance created".

### CODE REVIEW

**Comment by waheedahmed** [comment]
@awais786 @tasawernawaz @ahsan-ul-haq Please review this PR.

**Comment by waheedahmed** [comment]
@edx/ecommerce Please review this PR.

File: course_discovery/apps/publisher/emails.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    try:
+        object_path = reverse('publisher:publisher_course_run_detail', kwargs={'pk': course_run.id})
+        subject = _('Studio instance created')
+
+        to_addresses = course_run.course.get_group_users_emails()
+        from_address = settings.PUBLISHER_FROM_EMAIL
+
+        context = {
+            'course_run': course_run,
+            'run_page_url': 'https://{host}{path}'.format(
```
Comment (bderusha) [inline course_discovery/apps/publisher/emails.py]: `course_run_studio_page_url`

File: course_discovery/apps/publisher/emails.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+from django.core.urlresolvers import reverse
+from django.template.loader import get_template
+from django.utils.translation import ugettext_lazy as _
+
+
+logger = logging.getLogger(__name__)
+
+
+def send_email_for_studio_instance_created(course_run):
+    """ Send an email to course team on course key save.
```
Comment (awais786) [inline course_discovery/apps/publisher/emails.py]: Please update this doc string. Method name and doc string does not match and not explain the purpose of method.

File: course_discovery/apps/publisher/urls.py [hunk 3]
```diff
@@ -18,6 +18,11 @@
         views.ChangeStateView.as_view(),
         name='publisher_change_state'
     ),
+    url(
+        r'^course_runs/(?P<pk>\d+)/update/course_key/$',
```
Comment (bderusha) [inline course_discovery/apps/publisher/urls.py]: this should be a post request to `course_runs/(?P<pk>\d+)`.   

[RESTful Naming](http://www.restapitutorial.com/lessons/restfulresourcenaming.html)

**Review by bderusha — CHANGES_REQUESTED** [review]

File: course_discovery/apps/publisher/emails.py (hunk 1 above)
Comment (waheedahmed) [reply course_discovery/apps/publisher/emails.py]: @bderusha Its course run detail page url in publisher not for studio.

Comment (waheedahmed) [reply course_discovery/apps/publisher/emails.py]: changed to `course_run_page_url`.

**Review by awais786 — APPROVED** [review]

File: course_discovery/apps/publisher/emails.py (hunk 2 above)
Comment (waheedahmed) [reply course_discovery/apps/publisher/emails.py]: Done.

**Comment by waheedahmed** [comment]
@bderusha Please take another look.

**Comment by waheedahmed** [comment]
Closing this in favor of https://github.com/edx/course-discovery/pull/410.

**Pull request closed without merging**